src/legendre_polynomial.cpp

```diff
@@ -0,0 +1,83 @@
+#include <tallies/legendre_polynomial.hpp>
+
+#include <iostream>
+
+LegendrePolynomials::LegendrePolynomials(std::size_t order)
+    : legendre_coeff_(), order_(order) {
+  // evaluate the coefficient for each and every order
+  // from zero to given order
+
+  // get the total number of coefficient to store
+  std::size_t coeff_length = 1;
+  if (order_ != 0) {
+    if (order_ % 2 == 0) {
+      const std::size_t length =
+          static_cast<std::size_t>((static_cast<double>(order_) - 1.) * 0.5 + 1.);
+      coeff_length =
+          length * (length + 1) + static_cast<std::size_t>(static_cast<double>(order_) * 0.5 + 1);
+    } else {
+      const std::size_t length = static_cast<std::size_t>(static_cast<double>(order_) * 0.5 + 1);
+      coeff_length = length * (length + 1);
+    }
+  }
+  legendre_coeff_.reserve(coeff_length);
+
+  double inv_2 = 1.;
```

Comment:
Is this supposed to be 1. / 2. ? Also, I think this can be made const.

Reply:
I think, I should rename the variable from `double inv_2` to `double inv_2n` as this tells the inverse of the 2 to the power polynomial-order. The variable will be modified by multiplying 0.5 into for next order calculations. So, we cannot make it `const`.